android/cli/commands/_buildModule.js

```diff
@@ -61,21 +60,39 @@ AndroidModuleBuilder.prototype.migrate = function migrate(next) {
 	const cliSDKVersion = this.cli.sdk.manifest.version;
 	const manifestSDKVersion = this.manifest.minsdk;
 	const manifestModuleAPIVersion = this.manifest.apiversion;
-	const newVersion = semver.inc(this.manifest.version, 'major');
 	const manifestTemplateFile = path.join(this.platformPath, 'templates', 'module', 'default', 'template', 'android', 'manifest.ejs');
+	let newVersion = semver.inc(this.manifest.version, 'major');
 
 	var performMigration = function (next) {
 		this.logger.info(__('Migrating module manifest ...'));
 
+		// If a version is "1.0" instead of "1.0.0", semver currently fails. Work around it for now!
+		if (!newVersion) {
+			this.logger.warn(__('Detected non-semantic version (%s), will try to repair it!', this.manifest.version));
+			try {
+				newVersion = semver.inc(this.manifest.version.split('.')[0] + '.0.0', 'major');
+			} catch (e) {
+				this.logger.error(__('Unable to migrate version for you. Please update it manually by using a semantic version like "1.0.0" and try the migration again.'));
+				process.exit(1);
+			}
+		}
+
+		// Update the "apiversion" to the CLI API-version
 		this.logger.info(__('Setting %s to %s', 'apiversion'.cyan, cliModuleAPIVersion.cyan));
 		this.manifest.apiversion = cliModuleAPIVersion;
 
+		// Update the "minsdk" to the required CLI SDK-version
 		this.logger.info(__('Setting %s to %s', 'minsdk'.cyan, cliSDKVersion.cyan));
 		this.manifest.minsdk = cliSDKVersion;
 
+		// Update the "apiversion" to the next major
 		this.logger.info(__('Bumping version from %s to %s', this.manifest.version.cyan, newVersion.cyan));
 		this.manifest.version = newVersion;
 
+		// Add our new architecture
+		// TODO: Read this from a regex of the "manifestContent" instead to be more flexible
+		this.manifest.architectures = 'arm64-v8a ' + this.manifest.architectures;
```

Comment:
@janvennemann @ewanharris I really don't want this to be the final solution, so if you have some kind of clean JS way to detect & extract the new architectures from the manifest template, let me know! Unfortunately, I cannot think of a different place to grab them from.

Reply:
Maybe we should break them out into an entry in the android [package.json](https://github.com/appcelerator/titanium_mobile/blob/4ce94c7d7c0e75106dfc73a2ca04fd9c4420150d/android/package.json), we already have an array we define in the [_build.js](https://github.com/appcelerator/titanium_mobile/blob/93af11db2c37e3b64c7d748d7119932dc7cf1df6/android/cli/commands/_build.js#L76), plus the hardcoded values in the module template. So maybe having a single place would make sense vs 3 separate locations?

Reply:
Maybe a good idea. @garymathews thoughts?